protected/humhub/modules/content/components/ContentContainerDefaultPermissionManager.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2020 HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\content\components;
+
+use humhub\libs\BasePermission;
+use humhub\modules\user\components\PermissionManager;
+use humhub\modules\content\models\ContentContainerDefaultPermission;
+use Yii;
+
+/**
+ * @inheritdoc
+ */
+class ContentContainerDefaultPermissionManager extends PermissionManager
+{
+
+    /**
+     * @var string
+     */
+    public $contentcontainer_class = null;
+
+    /**
+     * @inheritdoc
+     */
+    protected function getModulePermissions(\yii\base\Module $module)
+    {
+        if ($module instanceof \humhub\components\Module) {
+            return $module->getPermissions(new $this->contentcontainer_class);
```

Comment:
Can't we just set the contentContainer here? Also note that `getPermissions` may return string classes instead of permission instances. In order to stay compatible current module behavior in which contentcontainer is optional?

Reply:
Ah, yes you are absolutely right, if we will change here the code `return $module->getPermissions(new $this->contentcontainer_class);` to:

```
$contentContainer = new $this->contentcontainer_class;
$permissions = $module->getPermissions($contentContainer);
foreach ($permissions as $permission) {
    if ($permission->contentContainer === null) {
        $permission->contentContainer = $contentContainer;
    }
}
return $permissions;
```

then we don't my migrations fixes in external modules.

Thank you for advice.
Sorry for the bad solution and what I did in all external modules.
Should I apply this new code and then revert all related changes in another modules?

Reply:
No problem, wasn't neccesarrily a bad solution. The purpose of `BasePermission::$contentContainer` is not clear on first sight and this is a reason we should get rid of it as mentioned in https://github.com/humhub/humhub/issues/4670#issuecomment-737865080

Reply:
If its not too much effort we should rather remove the migration, since the field may get deprecated soon.

Reply:
after implementation described in https://github.com/humhub/humhub/pull/4678#issuecomment-738632155 we may revert the initialization of `contentContainer` on each permission, because currently all query logic was moved into Permission Manager (`ContentContainerPermissionManager`) that already has an initialized Content Container.